mitiq/shadows/classical_postprocessing.py

```diff
@@ -8,130 +8,380 @@
-            obtained from ``random_pauli_measurement``.
-        pauli_str: Single observable consisting of Pauli operators.
-        k_shadows: Number of splits in the median of means estimator.
+        measurement_outcomes: A shadow tuple obtained from
+            `z_basis_measurement`.
+        pauli_str: Single mitiq observable consisting of
+            Pauli operators.
+        k_shadows: number of splits in the median of means estimator.
+        pauli_twirling_calibration: Whether to use Pauli twirling
+            calibration.
+        f_est: The estimated Pauli fidelities for each calibration
 
     Returns:
-        Estimation of the observable expectation value.
+        Float corresponding to the estimate of the observable
+        expectation value.
     """
-    # mitiq version
+    num_qubits = len(measurement_outcomes[0][0])
     obs = pauli_str._pauli
     coeff = pauli_str.coeff
 
     target_obs, target_locs = [], []
     for qubit, pauli in obs.items():
         target_obs.append(str(pauli))
         target_locs.append(int(qubit))
+    # if target_locs is [2,3,4], then target_support is "001110"
+    target_support = create_string(num_qubits, target_locs)
 
-    b_lists, u_lists = measurement_outcomes
-    u_lists = np.array([list(u) for u in u_lists])
-
-    n_total_measurements = len(b_lists)
+    # classical values stored in classical computer
+    b_lists_shadow = np.array(measurement_outcomes[0])
+    u_lists_shadow = np.array([list(u) for u in measurement_outcomes[1]])
+    # number of measurements in each split
+    n_total_measurements_shadow = len(b_lists_shadow)
     means = []
-
-    group_idxes = np.array_split(np.arange(n_total_measurements), k_shadows)
+    # loop over the splits of the shadow:
+    group_idxes = np.array_split(
+        np.arange(n_total_measurements_shadow), k_shadows
+    )
     # loop over the splits of the shadow:
     for idxes in group_idxes:
-        b_lists_k = b_lists[idxes]
-        u_lists_k = u_lists[idxes]
+        b_lists_shadow_k = b_lists_shadow[idxes]
+        u_lists_shadow_k = u_lists_shadow[idxes]
         # number of measurements/shadows in each split
-        n_group_measurements = len(b_lists_k)
-        # find the exact matches for the observable of
-        # interest at the specified locations
-        indices = np.all(u_lists_k[:, target_locs] == target_obs, axis=1)
-
-        # catch the edge case where there is no match in the chunk
-        if sum(indices) > 0:
-            # take the product and sum
-            product = np.prod(
-                3.0 * (b_lists_k[indices][:, target_locs]), axis=1
+        n_group_measurements = len(b_lists_shadow_k)
+
+        # observable is    obs = {IIXZYI}, or target_loc =[2,3,4],
+        # then the non-zero elements in product should satisfies (1) and (2) if
+        # calibration is used
+        # (1)shadow measure U_2 = {..XZY.}, exactly match
```

Comment:
This comment seems incomplete or maybe it is some old text to be removed.

Reply:
This is actually some comments on [conditions](https://github.com/unitaryfund/mitiq/blob/d6c2738da2fb506d903b587667a303c31d0cdef9/docs/source/examples/rshadows_tutorial.md?plain=1#L218-L221) in `rshadows_tutorial.md` 
$$
\hat{o}_\iota = \left(\sum_{b^{(1)}\in\{0,1\}^{n}}f_{b^{(1)}}^{-1}\prod_{j\in supp(O_\iota)}
\delta_{b_j^{(1)},1}\prod_{k\in supp(O_\iota)^c}\delta_{b_k^{(1)},0}\right)\prod_{i=1}^n  \langle b_i^{(2)}|U_i^{(2)}P_i U_i^{(2)\dagger}|b_i^{(2)}\rangle
$$
but I'll remove this, since it is very hard to explain clearly here in the comments.